### PR TITLE
Include Cart inventory on Checkout page

### PR DESCRIPTION
**Pull request by robinandeer**
## Describe your changes

Add cart inventory to `CheckoutPage`.

Generalize cart inventory components and move to "CartInventory" folder.

Move page form to handle sign inside `CheckoutPage` component.

Add new `CartCollapsible` component to `CheckoutPage`.

## Justify why they are needed

I will remove the old cart inventory components after we refactor `ConfirmationPage`.

## Checklist before requesting a review

- [ ] I have performed a self-review of my code

[GRW-1927]: https://hedvig.atlassian.net/browse/GRW-1927?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/store/src/components/CartInventory/CartEntryList.tsx [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  padding: 0,
+  listStyleType: 'none',
+  width: '100%',
+})
+StyledCartEntryList.defaultProps = { as: 'ul' }
+
+const HorizontalLineWithSpace = styled.hr(({ theme }) => ({
+  backgroundColor: theme.colors.gray300,
+  height: 1,
+  marginTop: theme.space[5],
```
Comment (alebedev) [inline apps/store/src/components/CartInventory/CartEntryList.tsx:23]: Nit: try `marginBlock: theme.space[5]` for both top and bottom margin.  Found by @guilhermespopolin :)

File: apps/store/src/components/CheckoutPage/CheckoutPage.tsx [hunk 2]
```diff
@@ -41,39 +88,46 @@ const CheckoutPage = (props: CheckoutPageProps) => {
...
-    </Space>
+      <HorizontalLineStandalone />
+    </Wrapper>
   )
 }
 
+const Wrapper = styled(Space)(({ theme }) => ({
+  paddingBottom: theme.space[6],
+}))
+Wrapper.defaultProps = { y: 1 }
```
Comment (gustaveen) [inline apps/store/src/components/CheckoutPage/CheckoutPage.tsx:129]: Just curios, why not inline the prop? :)

**Review by gustaveen — APPROVED** [review]
Nice one! 👍

File: apps/store/src/components/CheckoutPage/CartCollapsible/CartCollapsible.tsx [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+  const currencyFormatter = useCurrencyFormatter(cost.currencyCode)
+  const { t } = useTranslation(I18nNamespace.Checkout)
+
+  const [open, setOpen] = useState(false)
+  const closed = !open
+
+  return (
+    <Collapsible.Root open={open} onOpenChange={setOpen}>
+      <Trigger>
+        <SpaceBetween>
```
Comment (alebedev) [inline apps/store/src/components/CheckoutPage/CartCollapsible/CartCollapsible.tsx:28]: Can we use Space / SpaceFlex instead?

**Review by alebedev — APPROVED** [review]
Looks good.  Few questions inside

File: apps/store/src/components/CartInventory/CartEntryList.tsx (hunk 1 above)
Comment (guilhermespopolin) [reply apps/store/src/components/CartInventory/CartEntryList.tsx:23]: The nit thing is that `margin[inline|block]` is a logical value so it follows write mode and text orientation. We don't actually need that but it's good to know ℹ️

**Review by guilhermespopolin — APPROVED** [review]
Nicely done 👏🏻

**Comment by robinandeer** [comment]
**Graphite Merge Job**

Current status: 🔄 Waiting for checks to pass

This pull request is currently waiting for CI and mergeability checks to pass before being merged.

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/1085?utm_source=stack-comment).


Job Reference: TK6rKeYznaBPB65ZZFIH

**Pull request merged**